documentation/dt-bindings.md

```diff
@@ -0,0 +1,55 @@
+## Secure World OP-TEE DT bindings
+
+OP-TEE has a formal DT binding at
+
+```
+	firmware {
+		optee {
+		};
+	};
+```
+
+The Linux kernel device tree bindings documentation
+explains the properties that may appear there that
+are of interest to Normal World.
+
+OP-TEE may also be passed a DTB at boot-time, which
+may be different to, or modified from, the DTB passed
+to Normal World.
+
+The following properties inside /firmware/optee are
+understood by OP-TEE itself in the Secure World.
+
+### secure-device-id
+
+This allows the bootloader to declare an array of bytes
+which OP-TEE will use as the Secure Device ID.  For
+example OP-TEE uses this information to produce the
+key used for Secure Storage on that device.
+
+Since almost no SoC has public documentation on how to
+read the real device key data from e-fuses, this allows
+alternate ways to get per-device stable identity bits,
+eg, read the eMMC CID serial number on the same board.
+
+This may produce less data than a real key on the SoC,
+but it is still very useful since it requires no
+configuration to get a unique number.
```

Comment:
The problem with the eMMC CID is that it may be read from NS world too, which does not make it very good source of data to derive security keys... I do agree that it's convenient, so it's probably a valid example for some use cases, but I think you should state clearly that doing so is not much more secure than hard-coding a security key in the clear text binary.
Of course the bootloader may have access to other sources of private data, which really justifies this `secure-device-id` mechanism.